gateway/src/config_parser.rs

```diff
@@ -859,7 +859,7 @@ mod tests {
             .contains("Failed to parse config:\nunknown field `enable_agi`, expected"));
     }
 
-    /// Ensure that the config parsing fails when a JSON function has no output schema
+    /// Ensure that the config parsing defaults properly for JSON functions with no output schema
     #[test]
```

Comment:
We need a unit test somewhere that checks the default output schema works as expected (i.e. accepts any JSON and rejects everything else).